src/Server/Utils/Strategies/RegexResponseStrategy.php

```diff
@@ -67,13 +75,25 @@ private function bodyConditionIsRegex($expectation)
     private function fillWithUrlMatches($expectation, $httpRequest, $responseBody)
     {
         if ($this->urlConditionIsRegex($expectation)) {
-            $responseBody = preg_replace('/\$\{url\.(\d+)\}/', '\$$1', $responseBody);
-
-            return preg_replace(
-                $expectation->getRequest()->getUrl()->getValue(),
-                $responseBody,
-                $httpRequest->getUri()->__toString()
+            $pattern = preg_replace(
+                '/^(.)\^/',
+                '$1',
+                $expectation->getRequest()->getUrl()->getValue()
             );
+            preg_match(
```

Comment:
Could you please add $matches = []; before calling preg_match? It's always a good practice to preset the matches variable before calling preg_match.

Reply:
OK, I'll add it.